Library/Format.swift

```diff
@@ -323,13 +323,22 @@ private struct DateFormatterConfig {
 }
 
 extension DateFormatterConfig: Hashable {
-  fileprivate var hashValue: Int {
-    return
-      (self.template?.hashValue ?? 0)
-        ^ (self.dateStyle?.hashValue ?? 0)
-        ^ self.locale.hashValue
-        ^ (self.timeStyle?.hashValue ?? 0)
-        ^ self.timeZone.hashValue
+  fileprivate func hash(into hasher: inout Hasher) {
+    if let template = self.template {
```

Comment:
I'm not entirely sure about these...previously we would return `String` or `Int` which the `hasher.combine()` function doesn't like..the type should be known beforehand.

I assumed that adding `0` doesn't really affect the hash value but I might be wrong.